<launch>
  <arg name="operation_mode_transition_manager_param" default="$(find-pkg-share autoware_operation_mode_transition_manager)/config/operation_mode_transition_manager.param.yaml"/>
  <arg name="vehicle_info_param_file" default="$(find-pkg-share autoware_vehicle_info_utils)/config/vehicle_info.param.yaml"/>

  <!-- autoware_operation_mode_transition_manager -->
  <node pkg="autoware_operation_mode_transition_manager" exec="autoware_operation_mode_transition_manager_exe" name="autoware_operation_mode_transition_manager" output="screen">
    <!-- args -->
    <param from="$(var vehicle_info_param_file)"/>
    <param from="$(var operation_mode_transition_manager_param)"/>

    <!-- rename input topics -->
    <remap from="kinematics" to="/localization/kinematic_state"/>
    <remap from="steering" to="/vehicle/status/steering_status"/>
    <remap from="trajectory" to="/planning/scenario_planning/trajectory"/>
    <remap from="control_cmd" to="/control/command/control_cmd"/>
    <remap from="control_mode_report" to="/vehicle/status/control_mode"/>
    <remap from="gate_operation_mode" to="/control/vehicle_cmd_gate/operation_mode"/>
    <!-- from system -->

    <!-- rename output topics -->
    <remap from="is_autonomous_available" to="/control/is_autonomous_available"/>
    <remap from="control_mode_request" to="/control/control_mode_request"/>
    <!-- to vehicle -->
  </node>
</launch>
